pkg/util/k8s/labels.go

```diff
@@ -0,0 +1,31 @@
+package k8s
+
+import (
+	"log"
+	"regexp"
+)
+
+var labelAntiselector *regexp.Regexp
+
+func init() {
+	var err error
+
+	// Regex matching all non-allowed label characters
+	labelAntiselector, err = regexp.Compile("[^a-z0-9A-Z_.-]+")
```

Comment:
why not err := maybe?

Reply:
> why not err := maybe?

Won't work - `labelAntiselector` is global, so you can't use `:=`

Reply:
ic thx, is it the favorable method to use globals, or maybe a singleton is preferable ?

Reply:
It's actually a kind of singleton, since it's a private variable of a package.

Reply:
It is private and read-only within the package. It is as if it was a constant, except we need to populate it with a function in init.

Reply:
yep thanks, actually its almost the same as the pattern indeed
```
var instantiated *single = nil

func New() *single {
        if instantiated == nil {
                instantiated = new(single);
        }
        return instantiated;
}
```